File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/SkuIdFlattenedDto.java
package org.dddml.wms.domain;

import java.util.HashMap;
import java.util.Map;

public class SkuIdFlattenedDto
{


    public static final String[] propertyNames = new String[]{
            "productId",
            "attributeSetInstanceId",
        };

    public static final String[] propertyTypes = new String[]{
            "String",
            "String",
        };

    public static final Map<String, String> propertyTypeMap;

    static {
        propertyTypeMap = new HashMap<String, String>();
        for (int i = 0; i < propertyNames.length; i++ ) {
            propertyTypeMap.put(propertyNames[i], propertyTypes[i]);
        }
    }
	
    private SkuId value;

    public SkuIdFlattenedDto()
    {
        this(new SkuId());
    }

    public SkuIdFlattenedDto(SkuId value)
    {
        this.value = value;
    }

    public SkuId toSkuId()
    {
        return this.value;
    }


    public String getProductId()
    {
        return this.value.getProductId();
    }

    public void setProductId(String productId)
    {
        this.value.setProductId(productId);
    }

    public String getAttributeSetInstanceId()
    {
        return this.value.getAttributeSetInstanceId();
    }

    public void setAttributeSetInstanceId(String attributeSetInstanceId)
    {
        this.value.setAttributeSetInstanceId(attributeSetInstanceId);
    }


    @Override
    public boolean equals(Object obj)
    {
        if (obj == this) {
            return true;
        }
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        }

        SkuIdFlattenedDto other = (SkuIdFlattenedDto)obj;
        return value.equals(other.value);
    }

    @Override
    public int hashCode()
    {
        return value.hashCode();
    }

}

File: Dddml.Wms.HttpServices.PhpClient/Wms/Domain/SkuIdFlattenedDto.php
<?php

namespace Wms\Domain;

use JMS\Serializer\Annotation\Type;

class SkuIdFlattenedDto
{

    const PROPERTIES = [
            'productId' => 'string',
            'attributeSetInstanceId' => 'string',
        ];

    public static function getPropertyNames()
    {
        return array_keys(static::PROPERTIES);
    }

    public static function getPropertyTypes()
    {
        return array_values(static::PROPERTIES);
    }

	
    /**
     * @var SkuId
     */
    private $value;

    /**
     * @param SkuId $value
     */
    public function __construct(SkuId $value = null)
    {
        if ($value) {
            $this->value = $value;
        } else {
            $this->value = new SkuId();
        }
    }

    /**
     * @return SkuId
     */
    public function toSkuId()
    {
        return $this->value;
    }

    /**
     * @return string
     */
    public function getProductId()
    {
        return $this->value->getProductId();
    }

    /**
     * @param string $productId
     */
    public function setProductId($productId)
    {
        $this->value->setProductId($productId);
    }

    /**
     * @return string
     */
    public function getAttributeSetInstanceId()
    {
        return $this->value->getAttributeSetInstanceId();
    }

    /**
     * @param string $attributeSetInstanceId
     */
    public function setAttributeSetInstanceId($attributeSetInstanceId)
    {
        $this->value->setAttributeSetInstanceId($attributeSetInstanceId);
    }


}

